mne/preprocessing/_csd.py

```diff
@@ -0,0 +1,197 @@
+        out = inst
+
+    if picks is None:
+        picks = pick_types(inst.info, meg=False, eeg=True, exclude='bads')
+    elif not (isinstance(picks, list) or isinstance(picks, np.ndarray)):
+        raise ValueError('picks must be a np.ndarray or list, ' +
+                         ' got %s' % type(picks))
+
+    if len(picks) == 0:
+        raise ValueError('No EEG channels found.')
+
+    for name, matrix in dict(g_matrix=g_matrix, h_matrix=h_matrix).items():
+        if matrix is not None:
+            if (not isinstance(matrix, np.ndarray) or
+                    matrix.shape != (len(picks), len(picks))):
+                raise ValueError('An array that is shape n_channels x ' +
+                                 'n_channels is required for %s, got ' % name +
+                                 '%s %s' % (type(matrix), matrix.shape))
+
+    if lambda2 is None:
+        lambda2 = 1e-5
+
+    if (not (isinstance(lambda2, float) or isinstance(lambda2, int)) or
+            0 > lambda2 or lambda2 > 1):
+        raise ValueError('lambda2 must be a float less than 1, '
+                         'got %s' % lambda2)
+
+    if (not (isinstance(stiffness, float) or isinstance(stiffness, int)) or
+            stiffness < 0):
+        raise ValueError('stiffness must be non-negative numeric, '
+                         'got %s' % stiffness)
+
+    if not isinstance(n_legendre_terms, int) or n_legendre_terms < 1:
+        raise ValueError('n_legendre_terms must be an int greater than 0, '
+                         'got %s' % n_legendre_terms)
+
+    if (not (isinstance(head, float) or isinstance(head, int)) or
+            head <= 0):
+        raise ValueError('Head must be a float greater than 0, '
+                         'got %s' % head)
+
+    if not isinstance(n_jobs, int) or n_jobs < 1:
+        raise ValueError('n_jobs must be an int greater than 0, '
+                         'got %s' % n_jobs)
+
+    if not isinstance(copy, bool):
+        raise ValueError('copy must be True or False, got %s' % copy)
+
+    if ((g_matrix is None or h_matrix is None)):
+        pos = np.array([out.info['dig'][pick]['r'] for pick in picks])
+
+    G = g_matrix
+    if G is None:
+        G = _calc_g(np.dot(pos, pos.T), stiffness=stiffness,
+                    n_legendre_terms=n_legendre_terms)
+    H = h_matrix
+    if H is None:
+        H = _calc_h(np.dot(pos, pos.T), stiffness=stiffness,
+                    n_legendre_terms=n_legendre_terms)
+
+    G_precomputed = _prepare_G(G, lambda2)
+
+    if isinstance(out, BaseEpochs):
+        n_jobs = min(len(out), n_jobs)
+        logger.info('Using {} jobs'.format(n_jobs))
+        parallel, my_csd, _ = parallel_func(_compute_csd, n_jobs)
+        data = np.asarray(parallel(my_csd(e[picks],
+                                   G_precomputed=G_precomputed,
+                                   H=H, head=head) for e in out))
+        out.preload = True
+        out._data = data
+    elif isinstance(out, Evoked):
```

Comment:
you can just do `else` here, `evoked` has a `._data` wrapper for this purpose so you can do `inst._data[:] = `...